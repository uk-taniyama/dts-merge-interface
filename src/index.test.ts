import { existsSync, readFileSync, writeFileSync } from 'fs';

import { transformString } from '.';

describe('index', () => {
  const dataDir = `${__dirname}/../test-data/`;
  function test(name: string) {
    const inputPath = `${dataDir}/${name}.input.d.ts`;
    const expectedPath = `${dataDir}/${name}.output.d.ts`;
    const actualPath = `${dataDir}/${name}.actual.d.ts`;
    const input = readFileSync(inputPath, 'utf-8');
    const actual = transformString(input);
    if (!existsSync(expectedPath)) {
      writeFileSync(expectedPath, actual, 'utf-8');
      throw new Error(`save the actual data as the expected data.:${expectedPath}`);
    }
    const expected = readFileSync(expectedPath, 'utf-8');
    if (actual !== expected) {
      writeFileSync(actualPath, actual, 'utf-8');
      console.log('save actual', actualPath);
    }
    expect(actual).toEqual(expected);
  }

  it('transform', () => {
    test('test1');
  });
});
