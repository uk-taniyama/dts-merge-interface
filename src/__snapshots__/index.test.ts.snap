// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index transform: input 1`] = `
"declare module '.' {
    interface Xxx {
        children(): Xxx;
    }
}

declare const from: (name: string) => Xxx;
declare module '.' {
    interface Xxx {
        findTag(name: string): Xxx;
        findClass(name: string): Xxx;
    }
}

interface Yyy {
    foo(xxx: Xxx): number;
    bar(xxx: Xxx): number;
}
declare module '.' {
    interface Xxx {
        eq(index: number): Xxx;
    }
}

interface Xxx {
}

export { type Xxx, type Yyy, from };
"
`;

exports[`index transform: output 1`] = `
"interface Xxx {
    children(): Xxx;
    findTag(name: string): Xxx;
    findClass(name: string): Xxx;
    eq(index: number): Xxx;
}
interface Yyy {
    foo(xxx: Xxx): number;
    bar(xxx: Xxx): number;
}
declare const from: (name: string) => Xxx;
export { type Xxx, type Yyy, from };
"
`;
